drivers/FileSR.py

```diff
@@ -1082,6 +1102,49 @@ def _delete_cbt_log(self):
     def _cbt_log_exists(self, logpath):
         return util.pathexists(logpath)
 
+
+class SharedFileSR(FileSR):
+    """
+    FileSR subclass for SRs that use shared network storage
+    """
+    NO_HARDLINK_SUPPORT = "no_hardlinks"
+
+    def _raise_hardlink_error(self):
+        raise OSError(542, "Unknown error 524")
+
+    def _check_hardlinks(self):
+        test_name = os.path.join(self.path, str(uuid4()))
+        f = open(test_name, 'ab')
+        f.close
```

Comment:
`f.close()`

Or shorter:

`open(test_name, 'ab').close()`

Reply:
Re-running the test.